pytest_httpserver/httpserver.py

```diff
@@ -426,6 +426,20 @@ def match(self, request: Request) -> bool:
         return not difference
 
 
+def unpack_request_data(r: Request, auto_load: bool):
```

Comment:
I woudln't use this boolean parameter here and in the below code as it determines the function return type.
In 430 ther's an if which returns the data as-is, so I think this function should be made into two or more pieces. I don't like the auto-detecting thing. It would be better to require the caller to specify how it wants to decode the request.